src/BotPlutusInterface/Types.hs

```diff
@@ -277,6 +296,23 @@ instance Pretty LogLevel where
     Warn a -> "[WARNING " <> pretty a <> "]"
     Error a -> "[ERROR " <> pretty a <> "]"
 
+{- | if sufficientLogLevel settingLogLevel msgLogLvl
+ then message should be displayed with this log level setting.
+-}
+sufficientLogLevel :: LogLevel -> LogLevel -> Bool
+sufficientLogLevel logLevelSetting msgLogLvl =
+  msgLogLvl `constrLEq` logLevelSetting -- the log is important enough
+    && not (null intersectLogTypes) -- log is of type we're interested in
+  where
+    intersectLogTypes = ltLogTypes logLevelSetting `intersect` (ltLogTypes msgLogLvl <> [AnyLog])
+
+    toOrd (Error _) = 0 :: Int
```

Comment:
A bit of bike-shedding, I think we can eliminate the need to use toOrd by using `Data.Data` and `Enum`. Here's the complete code for that:
```
deriving stock instance Data LogType
deriving stock instance Data LogLevel

instance Enum LogLevel where
  fromEnum = (\a -> a - 1) . constrIndex . toConstr
  toEnum   = fromConstrB field . indexConstr (dataTypeOf $ Notice []) . (+ 1)

    where

      field :: forall a. Data a => a
      field = case eqT :: Maybe (a :~: [LogType]) of
          Just Refl -> [AnyLog]
          Nothing   -> error "Expected a value of type LogType."
```
in this way, if we add new data constructor for `LogLevel` we will not have to modify `toOrd`, and it's `Enum` instance will also get automatically handled for us.

The new `toOrd` will be:
```
toOrd a b = fromEnum a <= fromEnum b
```
For this you will need the following extension `{-# LANGUAGE DeriveDataTypeable #-}`.

It's just an idea, if you like it feel free to add it.

Reply:
oh yeah haha, thanks for the snippet, I'll put it tommorrow.

Only whats the reason for`+/- 1` in `to/fromEnum`?

Reply:
actually in `Data` instances the index of data constructors begins with 1, but in `Enum` it begins with 0. 
Eg: 
```
constrIndex (toConstr $ Error [AnyLog]) == 1
```
and so on... . So, just to be consistent we do (-1) from the index and return the value, and vice-versa.

Reply:
Included this and checked that ordering hasn't changed